packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useExportNoteAction.ts

```diff
@@ -0,0 +1,76 @@
+import { SingleRecordActionHookWithObjectMetadataItem } from '@/action-menu/actions/types/SingleRecordActionHook';
+import { recordStoreFamilyState } from '@/object-record/record-store/states/recordStoreFamilyState';
+import { BlockNoteEditor } from '@blocknote/core';
+import {
+  DOCXExporter,
+  docxDefaultSchemaMappings,
+} from '@blocknote/xl-docx-exporter';
+import {
+  PDFExporter,
+  pdfDefaultSchemaMappings,
+} from '@blocknote/xl-pdf-exporter';
+import * as ReactPDF from '@react-pdf/renderer';
+import { Buffer } from 'buffer';
+import { Packer } from 'docx';
+import { saveAs } from 'file-saver';
+import { useRecoilValue } from 'recoil';
+import { isDefined } from 'twenty-ui';
+
+// Polyfill needed for exportNoteToDocX
+if (typeof window !== 'undefined') {
+  window.Buffer = Buffer;
+}
+
+const exportNoteToPdf = async (editor: BlockNoteEditor, filename: string) => {
+  const exporter = new PDFExporter(editor.schema, pdfDefaultSchemaMappings);
+
+  const pdfDocument = await exporter.toReactPDFDocument(editor.document);
+
+  const blob = await ReactPDF.pdf(pdfDocument).toBlob();
+  saveAs(blob, `${filename}.pdf`);
+};
+
+const exportNoteToDocx = async (editor: BlockNoteEditor, filename: string) => {
+  const exporter = new DOCXExporter(editor.schema, docxDefaultSchemaMappings);
+
+  const docxDocument = await exporter.toDocxJsDocument(editor.document);
+
+  const blob = await Packer.toBlob(docxDocument);
+  saveAs(blob, `${filename}.docx`);
+};
+
+export const useExportNoteAction: SingleRecordActionHookWithObjectMetadataItem =
+  ({ recordId, objectMetadataItem }) => {
+    const selectedRecord = useRecoilValue(recordStoreFamilyState(recordId));
+
+    const filename = `${selectedRecord?.title || 'Untitled Note'}`;
+
+    const isNoteOrTask =
+      objectMetadataItem?.nameSingular === 'note' ||
```

Comment:
CoreObjectNameSingular.Note // CoreObjectNameSingular.Task

Reply:
Resolved in recent commit.